nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -292,7 +293,10 @@ private static boolean isDirectTypeUseAnnotation(Attribute.TypeCompound t) {
     // location is a list of TypePathEntry objects, indicating whether the annotation is
     // on an array, inner type, wildcard, or type argument. If it's empty, then the
     // annotation is directly on the type.
-    return t.position.location.isEmpty();
+    // We care about both annotations directly on the outer type and also those on directly
+    // on an inner type, but wish to discard annotations on arrays, wildcards, or type arguments.
+    return t.position.location.stream()
+        .allMatch(l -> l.tag.equals(TypeAnnotationPosition.TypePathEntryKind.INNER_TYPE));
```

Comment:
Why do we need `allMatch` here?  Don't we want to check that there _exists_ an element of the list with tag `INNER_TYPE` such that it is on the "innermost" type?  E.g., what if we have field declaration `A.@Nullable B.C foo;`; do we want to treat that as a `@Nullable` reference?

Also we may have a type like `A.@Nullable B<@Nullable C>`.  I think we should treat this as as a `@Nullable` reference, but `allMatch` may be too strict?  I'm not sure if we'll get a single `TypeCompound` in two different locations for that case...

Reply:
Upon further investigation, I _think_ `location` only has length greater than 1 for multi-dimensional arrays.  For a type like `A.@Nullable B<@Nullable C>` we will get two different `TypeCompound` objects each of whose `position.location` lists will be length 1.

Reply:
We do need `allMatch`, because `A.B<@Nullable C>` shows as having the location array `["INNER_TYPE", "TYPE_ARGUMENT(0)"]`.

Think of it like a path: "annotation found at the type argument, position 0, of the inner type".

This is needed, because otherwise stuff like `A<@Bar E>.B<C>.` would be ambiguous if `@Bar` just says `["TYPE_ARGUMENT(0)"]`.

So, in your example above, if we see `A.@Nullable B<@Nullable C>`, what we get is two `@Nullable` annotations (two calls to `getTypeUseAnnotations`) one which has the location array  `["INNER_TYPE", "TYPE_ARGUMENT(0)"]` (the one by `C`) and one which has location array `["INNER_TYPE"]`. We want to match only the second.

We want to match "a path including only a chain of inner types" and that's `allMatch`.